Emby.Server.Implementations/Playlists/PlaylistManager.cs

```diff
@@ -189,35 +189,42 @@ public void AddToPlaylist(string playlistId, IEnumerable<Guid> itemIds, Guid use
 
         private void AddToPlaylistInternal(string playlistId, IEnumerable<Guid> itemIds, User user, DtoOptions options)
         {
-            var playlist = _libraryManager.GetItemById(playlistId) as Playlist;
-
-            if (playlist == null)
-            {
-                throw new ArgumentException("No Playlist exists with the supplied Id");
-            }
-
-            var list = new List<LinkedChild>();
+            // Retrieve the existing playlist
+            var playlist = _libraryManager.GetItemById(playlistId) as Playlist
+                ?? throw new ArgumentException("No Playlist exists with Id " + playlistId);
 
+            // Retrieve all the items to be added to the playlist
             var items = GetPlaylistItems(itemIds, playlist.MediaType, user, options)
                 .Where(i => i.SupportsAddingToPlaylist)
                 .ToList();
 
-            foreach (var item in items)
+            // Remove duplicates from the new items to be added
+            var existingIds = playlist.LinkedChildren.Select(c => c.ItemId).ToHashSet();
+            var uniqueItems = items
+                .Where(i => !existingIds.Contains(i.Id))
+                .GroupBy(i => i.Id)
+                .Select(group => group.First())
```

Comment:
Maybe it's worth creating an `IEqualityComparer` to `Distinct` `BaseItem`s by Id?

Reply:
I have a branch that adds `Equals()` to `BaseItem`, let me fire a PR for that :)

Reply:
https://github.com/jellyfin/jellyfin/pull/2502

Reply:
@mark-monteiro so this pull request is waiting on that one then?

Reply:
@dkanada yes, and we decided not to merge that one until after the release. If this is something we really want to get into the release I could implement `IEqualityComparer` here, then remove that later once #2505 is merged.

Reply:
Updated this to use `Distinct()` with the new equality/hashcode implementation for `BaseItem`